src/read.rs

```diff
@@ -228,6 +230,50 @@ impl<R: Read+io::Seek> ZipArchive<R>
         })
     }
 
+    /// Extract a Zip archive into a directory.
+    ///
+    /// Paths are sanitized so that they cannot escape the given directory.
+    ///
+    /// # Platform-specific behaviour
+    ///
+    /// On unix systems permissions from the zip file are preserved, if they exist.
+    pub fn extract(&mut self, directory: &Path) -> ZipResult<()> {
+        for i in 0..self.len() {
+            let mut file = self.by_index(i)?;
+            let filepath = file.sanitized_name();
+
+            // `sanitized_name` should return a relative path
+            // otherwise there's a risk of directory traversal attacks
+            assert!(filepath.is_relative());
+
+            let outpath = directory.join(filepath);
+
+            if (&*file.name()).ends_with('/') {
+                fs::create_dir_all(&outpath)?;
+            } else {
+                if let Some(p) = outpath.parent() {
```

Comment:
Can we use `matches!` instead of nested `if let` here? 
```rust
matches!(outpath.parent(), Some(p) if p.exists)
```

Reply:
`matches!` is pretty recent (1.42.0, 2020-03-12), I'm not sure it's worth bumping the MSRV just for this?

Reply:
Good point. We don't yet have a MSRV policy yet, but we should establish one